sql/spanner/inspect_test.go

```diff
@@ -0,0 +1,281 @@
+			require.NoError(t, err)
+			mk := mock{m}
+			mk.databaseOpts(dialectGoogleStandardSQL)
+			var drv migrate.Driver
+			drv, err = Open(db)
+			require.NoError(t, err)
+			mk.ExpectQuery(sqltest.Escape(schemasQueryArgs)).
+				WithArgs([]string{""}).
+				WillReturnRows(sqlmock.NewRows([]string{"schema_name"}).AddRow(""))
+			tt.before(mk)
+			s, err := drv.InspectSchema(context.Background(), "", nil)
+			require.NoError(t, err)
+			tt.expect(require.New(t), s.Tables[0], err)
+		})
+	}
+}
+
+func TestDriver_Realm(t *testing.T) {
+	db, m, err := sqlmock.New(sqlmock.ValueConverterOption(mockValueConverter{}))
+	require.NoError(t, err)
+	mk := mock{m}
+	mk.databaseOpts(dialectGoogleStandardSQL)
+	drv, err := Open(db)
+	require.NoError(t, err)
+	mk.ExpectQuery(sqltest.Escape(schemasQuery)).
+		WillReturnRows(sqlmock.NewRows([]string{"schema_name"}).AddRow(""))
+	m.ExpectQuery(sqltest.Escape(tablesQuery)).
+		WithArgs([]string{""}).
+		WillReturnRows(sqlmock.NewRows([]string{"table_schema", "table_name", "parent_table_name", "on_delete_action", "spanner_state"}))
+	realm, err := drv.InspectRealm(context.Background(), &schema.InspectRealmOption{})
+	require.NoError(t, err)
+	require.EqualValues(t, func() *schema.Realm {
+		r := &schema.Realm{
+			Schemas: []*schema.Schema{
+				{
+					Name: "default",
+				},
+			},
+		}
+		r.Schemas[0].Realm = r
+		return r
+	}(), realm)
+
+	mk.ExpectQuery(sqltest.Escape(schemasQueryArgs)).
+		WithArgs([]string{""}).
+		WillReturnRows(sqlmock.NewRows([]string{"schema_name"}).AddRow(""))
+	m.ExpectQuery(sqltest.Escape(tablesQuery)).
+		WithArgs([]string{""}).
+		WillReturnRows(sqlmock.NewRows([]string{"table_schema", "table_name", "parent_table_name", "on_delete_action", "spanner_state"}))
+	realm, err = drv.InspectRealm(context.Background(), &schema.InspectRealmOption{Schemas: []string{""}})
+	require.NoError(t, err)
+	require.EqualValues(t, func() *schema.Realm {
+		r := &schema.Realm{
+			Schemas: []*schema.Schema{
+				{
+					Name: "default",
+				},
+			},
+		}
+		r.Schemas[0].Realm = r
+		return r
+	}(), realm)
+}
+
+func TestInspectMode_InspectRealm(t *testing.T) {
+	db, m, err := sqlmock.New(sqlmock.ValueConverterOption(mockValueConverter{}))
+	require.NoError(t, err)
+	mk := mock{m}
+	mk.databaseOpts(dialectGoogleStandardSQL)
+	mk.ExpectQuery(sqltest.Escape(schemasQuery)).
+		WillReturnRows(sqlmock.NewRows([]string{"schema_name"}).AddRow(""))
+	drv, err := Open(db)
+	realm, err := drv.InspectRealm(context.Background(), &schema.InspectRealmOption{Mode: schema.InspectSchemas})
+	require.NoError(t, err)
+	require.EqualValues(t, func() *schema.Realm {
+		r := &schema.Realm{
+			Schemas: []*schema.Schema{
+				{
+					Name: "default",
+				},
+			},
+		}
+		r.Schemas[0].Realm = r
+		return r
+	}(), realm)
+}
+
+const dialectGoogleStandardSQL = "GOOGLE_STANDARD_SQL"
+
+type mock struct {
+	sqlmock.Sqlmock
+}
+
+func (m mock) CheckNamedValue(value *driver.NamedValue) (err error) {
```

Comment:
Can you elaborate on why we need this? I think you can

Reply:
I/we are getting errors ala:
`InspectTables: issue in tables(): query tables: sql: converting argument $1 type: unsupported type []string, a slice of string` though maybe we can rearrange our information_schema queries to avoid the need for the slice input.

Reply:
Yes, but let's do it in future PRs.